packages/gitlab/src/api/GitlabCIClient.ts

```diff
@@ -213,7 +215,7 @@ export class GitlabCIClient implements GitlabCIApi {
     ): Promise<ContributorsSummary | undefined> {
         const contributorsData = await this.callApi<PeopleCardEntityData[]>(
             'projects/' + projectID + '/repository/contributors',
-            { sort: 'desc' }
+            { sort: 'desc', without_project_bots: true }
```

Comment:
This API does not have this parameter. This will not work. To give you more context, In the last Gitlab version, contributors were automatically filtered, but the old versions will not be filtered. We have support for also old gitlab version when it is possible. Then, For this task, you can simply add the `without_project_bots` to the `users` call without modifying the mocked data.

Reply:
You have to ensure that with this parameter, bots are not showed

Reply:
Thanks for the feedback, will work on this today :smile: I see what you mean.

Reply:
Just want to double-check @antoniomuso you do mean this method? `getUserDetail` because both this one and `getUserProfilesData` calls the user's endpoint (albeit with different query params)? So an option would also be to do this?

This would mean any call to the /users endpoint would include `without_project_bots`

```TypeScript
protected async callApi<T>(path: string, query: { [key in string]: any }): Promise<T | null> {
    const apiUrl = `${await this.discoveryApi.getBaseUrl('gitlab')}/${this.gitlabInstance}`;

    // Add the 'without_project_bot' parameter to the query object if the path is '/users'
    if (path === 'users') {
        query.without_project_bots = 'true';
    }

    const response = await fetch(`${apiUrl}/${path}?${new URLSearchParams(query).toString()}`);

    // Handle the response as before...
}
```

Or we just add it to `getUserDetail` like this:

```TypeScript
private async getUserDetail(username: string): Promise<PeopleCardEntityData> {
    if (username.startsWith('@')) {
        username = username.slice(1);
    }

    const query = {
        username,
        without_project_bots: 'true',
    };

    const userDetail = (await this.callApi<PeopleCardEntityData[]>('users', query))?.[0];

    if (!userDetail) {
        throw new Error(`user ${username} does not exist`);
    }

    return userDetail;
}
```
Apologies if I misunderstood, but maybe you can point me in the right direction :smile, thanks a lot, and looking forward to working with you guys :smile:

Reply:
It is better the second way but applied only to the method `getUserProfilesData`, which is the method used by `getContributorsSummary`.